reporter/src/main/kotlin/reporters/StaticHtmlReporter.kt

```diff
@@ -514,7 +514,10 @@ class StaticHtmlReporter : Reporter {
                                 val firstFinding = finding.locations.first()
 
                                 val permalink = when {
-                                    row.vcsInfo != VcsInfo.EMPTY -> {
+                                    // TODO: As we do not have access to provenance information here anymore (see issue
+                                    //       #2631), use a "META-INF" path prefix as a crude hint whether a Java source
+                                    //       artifact was scanned.
+                                    row.vcsInfo != VcsInfo.EMPTY && !firstFinding.path.startsWith("META-INF") -> {
```

Comment:
As the check only works if the "randomly selected" first finding is in the `META-INF` directory, it can be improved to at least work for all scans which do have a `META-INF` directory underneath the root directory with a license finding in it, like: `val isSourceArtifact = row.detectedLicenses.any { it.locations.any { it.path.startWith("META-INF") } }`
What do you think?

Reply:
Done.